libs/vkd3d/state.c

```diff
@@ -3287,6 +3308,11 @@ static bool d3d12_pipeline_state_can_use_dynamic_stride(struct d3d12_pipeline_st
                   graphics->minimum_vertex_buffer_dynamic_stride[slot]);
             return false;
         }
+        else if (dyn_state->vertex_strides[slot] & graphics->vertex_buffer_stride_align_mask[slot])
+        {
+            TRACE("Stride alignment does not conform to D3D12. Falling back to static pipeline.\n");
+            return false;
```

Comment:
can't we just fix strides at bind time rather than compiling an entirely new pipeline?

Reply:
I guess we would modify the strides like we modify VBO offsets.

Reply:
Yeh, I'd prefer doing that instead since there doesn't really seem to be any need to make the strides static.

Reply:
Fixed.